agent/consul/server.go

```diff
@@ -1552,8 +1553,29 @@ func (s *Server) RPC(ctx context.Context, method string, args interface{}, reply
 		metrics.IncrCounter([]string{"client", "rpc", "exceeded"}, 1)
 		return structs.ErrRPCRateExceeded
 	}
+
+TRY:
 	if err := s.rpcServer.ServeRequest(codec); err != nil {
-		return err
+		info, _ := args.(structs.RPCInfo)
+
+		retryableMessages := []error{
+			// These rate limit errors are returned before the handler is called, so are
+			// safe to retry.
+			rpcRate.ErrRetryElsewhere,
+		}
+
+		if retry := canRetry(info, err, firstCheck, s.config, retryableMessages); !retry {
+			metrics.IncrCounterWithLabels([]string{"client", "rpc", "failed"}, 1, []metrics.Label{{Name: "server", Value: s.config.NodeName}})
```

Comment:
The metrics error we are getting is because of this added metric call which is not tested, but I think with the comment above around no need to retry it should go away.